src/module/ct/ct.rs

```diff
@@ -0,0 +1,63 @@
+use anyhow::{bail, Result};
+
+use super::{bpf::CtEventFactory, ct_hook};
+use crate::{
+    cli::{dynamic::DynamicCommand, CliConfig},
+    collect::Collector,
+    core::{
+        events::EventSectionFactory,
+        inspect,
+        probe::{Hook, ProbeManager},
+    },
+    module::{Module, ModuleId},
+};
+
+pub(crate) struct CtModule {
+    // Whether the event capturing module was initialized
+    init: bool,
+}
+
+impl Collector for CtModule {
+    fn new() -> Result<Self> {
+        Ok(CtModule { init: false })
+    }
+
+    fn known_kernel_types(&self) -> Option<Vec<&'static str>> {
+        Some(vec!["struct sk_buff *"])
+    }
+
+    fn register_cli(&self, cmd: &mut DynamicCommand) -> Result<()> {
+        cmd.register_module_noargs(ModuleId::Ct)
+    }
+
+    fn can_run(&mut self, _cli: &CliConfig) -> Result<()> {
+        let kernel = &inspect::inspector()?.kernel;
+
+        if let (Ok(None), Ok(None)) = (
+            kernel.get_config_option("CONFIG_NF_CONNTRACK"),
+            kernel.get_config_option("CONFIG_NF_CONNTRACK_MODULE"),
```

Comment:
Oh OK. Sorry it took long to get it. Yes, it's not trivial from the function definition so it would be nice to have a helper or an enum.

Reply:
Also, what if neither kconfigs exist. Shouldn't we fail as well?

Reply:
What I mean is that  CONFIG_NF_CONNTRACK_MODULE seems redundant if we anyhow need to check the module is loaded

Reply:
So maybe we need something like this:
```rust
        match kernel.get_config_option("CONFIG_NF_CONNTRACK") {
            Ok(Some("y")) => (),
            Ok(Some("m")) => {
                if kernel.is_module_loaded("nf_conntrack") != Some(false) {
                    bail!("'nf_conntrack' module is not loaded");
                }
            }
            Err(_) => (), // Don't fail if kconfig is not available. Try our best.
            _ => bail!("This kernel does not support connection tracking"),
        }

```

Reply:
> 
> 
> Yes, that should work[1].
> 
> [1] if kernel.is_module_loaded("nf_conntrack") != Some(false) should be if kernel.is_module_loaded("nf_conntrack") == Some(false)
> 

Yes, I edited it, probably just after you loaded the page :-D

Also, regarding our attempt not to depend  on kconfig being present.
Is there a warning about that? Something like "Kernel configuration is not available, collectors might not fully work"?
Because I'd be very confused if the tool does not say anything and some data stuff is just missing

Reply:
> So maybe we need something like this:
> 
> ```rust
>         match kernel.get_config_option("CONFIG_NF_CONNTRACK") {
>             Ok(Some("y")) => (),
>             Ok(Some("m")) => {
>                 if kernel.is_module_loaded("nf_conntrack") == Some(false) {
>                     bail!("'nf_conntrack' module is not loaded");
>                 }
>             }
>             Err(_) => (), // Don't fail if kconfig is not available. Try our best.
>             _ => bail!("This kernel does not support connection tracking"),
>         }
> ```

Ack to this. Just reporting what we said offline:

other than redundant, CONFIG_NF_CONNTRACK_MODULE is not an actual kconfig, so it's good to get rid of it.